### PR TITLE
Update NCO scheduling app

### PR DESCRIPTION
**Pull request by thompscs**
## Overview: ##

Updates to the NCO scheduling app main page located at https://designsafe-ci.org/nco to replace the prototype design and the fake sample dataset with a simple set of real project data from actual grants.  Boilerplate text and links relevant to facility scheduling is added to clarify the purpose for users as well as additional columns and updated filters & sorts within the data table.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* n/a

## Summary of Changes: ##

* Adding text to app's primary page.
* Changing styling to match other areas of the DS website.
* Switching the data table to pull from a set of real project data.
* Updating Angular components of the data table and python code to support additional columns for the real dataset.
* Updating the filters and sorts to match the real datasets.

## Testing Steps: ##
1. Visit the app page at /nco
2. Confirm the table is loading data from the correct dataset.  If there are rows that show NSF grant numbers and PI names anywhere in the table, then it is loading from the new dataset.
3. Try the new filters and sorts.

## UI Photos:

Screenshot from 2021-08-12 23-44-20

Screenshot from 2021-08-05 11-22-24

Screenshot from 2021-08-05 11-22-38

## Notes: ##

The goal of this page is to eventually support a much wider set of per-experiment level data, but that does not yet exist.  The commercial tool being used to schedule projects during the first award period did not collect the type of data we want to show here.  So, for the near future this page will be much closer to the simple prototype design until enough data with higher granularity is collected in projects moving forward to allow us to expand it.

### CODE REVIEW

**Review by fnets — COMMENTED** [review]
The UI looks good to me, but since we don't have data to test with, it's hard to test test that this actually works. Is there dummy data available?

**Comment by thompscs** [comment]
> The UI looks good to me, but since we don't have data to test with, it's hard to test test that this actually works. Is there dummy data available?

The data collection it will use isn't completely populated yet, but it is already live with a couple dozen items.  Both the prototype page that's on the production site right now and this updated version are using the same mongo db, just diverted to a different collection temporarily ("projects_v2" vs "projects").

I didn't want to pull the rug from under  the prototype page live right now on the production server by changing the format of its collection underneath it.  Also, we haven't finalized what all we will need in the new version.  The dataset we have now is very limited as our old scheduling system didn't store per-event dates or status, only grant-wide start/stop info.  I created the "projects_v2" version for now with the goal of switching back to just "projects" in the future once the current page after we see how much extra data on projects the facilities are willing to start collecting for us, to have a better idea of what the final mongo document format will be.

1. Prototype page is live now to the public using "projects".
2. This updated version uses "projects_v2", which already exists and will be continue to be filled overtime with our historical data.
3. After the current prototype page is gone, we can get rid of the dummy data that is is in "projects" now.
4. After we start getting new format of data from facilities and settle our battles with them over how much they are willing to provide, we figure out best format moving forward and repopulate "projects" with the new style documents.
5. Future version of this page switches back to new and improved "projects" and we either reimport the "projects_v2" entries  into the new format or keep it around as-is, as a sort of "archive" collection of older grants that can be browsed separately.

File: designsafe/static/scripts/nco/components/nco-scheduler-list/nco-scheduler-list.template.html [hunk 1]
```diff
@@ -2,16 +2,29 @@
...
+	    <td class="event-action">
+	       <div class="a-fac"><span><a href="mailto:zehner2@purdue.edu?subject=Requesting Contact with {{event.facility}}&body=Contact Request for {{event.facility}}%0D%0A%0D%0APlease provide details about why you would like to be placed in contact with the {{event.facility}} experimental facility and how best to contact you if not via a reply to this email address.%0D%0A%0D%0A--------------------%0D%0A%0D%0A">Contact Facility</a></span></div>
+	      <div class="a-payload"><span><a href="mailto:zehner2@purdue.edu?subject=Payload Request for Award #{{event.awardNumber || ' N/A'}}&body=PAYLOAD REQUEST%0D%0A%0D%0AProject: {{event.awardTitle}}%0D%0AAward #{{event.awardNumber || ' N/A'}}%0D%0AFacility: {{event.facility}}%0D%0AStart Date: {{event.dateStart}}%0D%0AEnd Date: {{event.dateEnd}}%0D%0APlease provide details about your payload project for the award listed above and how best to contact you.%0D%0A%0D%0A--------------------%0D%0A%0D%0A">Request Payload</a></span></div>
+	       <!-- TODO: create a generic FSOC forwarding email address or load it from a variable -->
+	    </td>
```
Comment (rstijerina) [inline designsafe/static/scripts/nco/components/nco-scheduler-list/nco-scheduler-list.template.html:27]: Formatting
```suggestion
            <td class="event-title-prj">
                <!-- we want to show both, but not enough data points have both yet.
                <p class="event-title">{{event.eventTitle}}</p>
                <p class="project-title">Award Title</p> -->
                <p class="event-title">{{event.awardTitle}}</p>
                <p data-ng-if="event.awardNumber"><a class="nsf"
                        href="https://www.nsf.gov/awardsearch/showAward?AWD_ID={{event.awardNumber}}">NSF
                        #{{event.awardNumber}}</a></p>
                <span class="pi-name">PI: {{event.piName || 'not reported'}}</span>
            </td>
            <td class="event-desc">{{event.awardDescription || 'N/A' }}</td>
            <td class="event-action">
                <div class="a-fac"><span><a
                            href="mailto:zehner2@purdue.edu?subject=Requesting Contact with {{event.facility}}&body=Contact Request for {{event.facility}}%0D%0A%0D%0APlease provide details about why you would like to be placed in contact with the {{event.facility}} experimental facility and how best to contact you if not via a reply to this email address.%0D%0A%0D%0A--------------------%0D%0A%0D%0A">Contact
                            Facility</a></span></div>
                <div class="a-payload"><span><a
                            href="mailto:zehner2@purdue.edu?subject=Payload Request for Award #{{event.awardNumber || ' N/A'}}&body=PAYLOAD REQUEST%0D%0A%0D%0AProject: {{event.awardTitle}}%0D%0AAward #{{event.awardNumber || ' N/A'}}%0D%0AFacility: {{event.facility}}%0D%0AStart Date: {{event.dateStart}}%0D%0AEnd Date: {{event.dateEnd}}%0D%0APlease provide details about your payload project for the award listed above and how best to contact you.%0D%0A%0D%0A--------------------%0D%0A%0D%0A">Request
                            Payload</a></span></div>
                <!-- TODO: create a generic FSOC forwarding email address or load it from a variable -->
            </td>
```

File: designsafe/apps/nco/templates/designsafe/apps/nco/nco_index.j2 [hunk 2]
```diff
@@ -2,19 +2,88 @@
...
+              <div style="padding-top: 10px;">
+                <p><b><a href="/facilities/nco/contact/">Questions? Contact Us</a></b></p>
+              </div>
+            </div>
+
+	  </div>
+	</div>
+	<hr class="top-line"/>
+
+	<!-- SCHEDULER VIEW -->
```
Comment (rstijerina) [inline designsafe/apps/nco/templates/designsafe/apps/nco/nco_index.j2:76]: Formatting
```suggestion
        <!-- PAGE INTRO ROW -->
        <div class="top-row">
            <!-- SCHEDULER INTRO -->
            <div class="title-intro-box">
                <h1> Facility Scheduling Dashboard </h1>
                <h3><em>Centralized management for NSF-funded NHERI grant awards</em></h3>
                <p>Welcome to the NHERI Facility Scheduling Dashboard.</p>
                <p>At any given time, dozens of research projects are underway at NHERI's <a
                        href="https://www.designsafe-ci.org/facilities/experimental/">eight Experimental Facilities</a>. The
                    Dashboard providers researchers with a centralized location for managing project experiments and offers
                    a public view of the work underway at NHERI facilities.</p>
                <p>The NHERI <a href="https://www.designsafe-ci.org/facilities/nco">Network Coordination Office (NCO)</a>
                    oversee's NHERI's centralized scheduling. Questions? Contact <a href="mailto:zehner2@purdue.edu"
                        target="_blank">Dan Zehner</a>, NCO Facility Scheduling and Operations Coordinator.</p>
            </div>
            <!-- TOP ACTION BOX -->
            <div class="top-action-box">
                <div style="clear: both;">
                    <div class="box-topper box-topper-ef text-center top-action-icon"><i
                            class="ds-icon ds-icon-toolbox"></i></div>
                    <div style="padding-top: 10px;">
                        <p><b><a href="/facilities/scheduling/request/">Request a New Project</a></b></p>
                    </div>
                </div>

                <div style="clear: both;">
                    <div class="box-topper box-topper-ef text-center top-action-icon"><i
                            class="ds-icon ds-icon-toolbox"></i></div>
                    <div style="padding-top: 10px;">
                        <p><b><a href="/facilities/scheduling/protocol/">NHERI Scheduling Protocol</a></b></p>
                    </div>
                </div>

                <div style="clear: both;">
                    <div class="box-topper box-topper-ef text-center top-action-icon"><i
                            class="ds-icon ds-icon-toolbox"></i></div>
                    <div style="padding-top: 10px;">
                        <p><b><a href="/facilities/">NHERI Experimental Facilities</a></b></p>
                    </div>
                </div>

                <div style="clear: both;">
                    <div class="box-topper box-topper-ef text-center top-action-icon"><i
                            class="ds-icon ds-icon-toolbox"></i></div>
                    <div style="padding-top: 10px;">
                        <p><b><a href="/facilities/nco/contact/">Questions? Contact Us</a></b></p>
                    </div>
                </div>

            </div>
        </div>
        <hr class="top-line" />

        <!-- SCHEDULER VIEW -->
```

File: designsafe/apps/nco/templates/designsafe/apps/nco/nco_index.j2 [hunk 3]
```diff
@@ -2,19 +2,88 @@
...
         <nco-scheduler>
             <div class="bg-info text-info" data-ng-if="!$ctrl.initialized">
                 <i class="fa fa-spinner fa-spin">&nbsp;</i> Loading...
             </div>
         </nco-scheduler>
+	
+	<!-- CLOSING ROW -->
+	<hr class="mid-line"/>
+        <h3>About the NHERI Network Coordination Office</h3>
+        <p>The NCO's centralized facility oversight and coordination allows for fair and transparent use of these NSF-funded Experimental Facilities following a scheduling protocol developed in cooperation with facilities.  The <a href="https://www.designsafe-ci.org/facilities/nco">NHERI NCO</a> developed and currently maintains the NHERI Facility Scheduling Dashboard as part of its administrative and governance roles within the NHERI network.  The NHERI scheduling protocol is available <a href="https://www.designsafe-ci.org/facilities/scheduling/protocol">here</a>.</p>
```
Comment (rstijerina) [inline designsafe/apps/nco/templates/designsafe/apps/nco/nco_index.j2:86]: Formatting
```suggestion
        <!-- CLOSING ROW -->
        <hr class="mid-line" />
        <h3>About the NHERI Network Coordination Office</h3>
        <p>The NCO's centralized facility oversight and coordination allows for fair and transparent use of these NSF-funded
            Experimental Facilities following a scheduling protocol developed in cooperation with facilities. The <a
                href="https://www.designsafe-ci.org/facilities/nco">NHERI NCO</a> developed and currently maintains the
            NHERI Facility Scheduling Dashboard as part of its administrative and governance roles within the NHERI network.
            The NHERI scheduling protocol is available <a
                href="https://www.designsafe-ci.org/facilities/scheduling/protocol">here</a>.</p>
```

**Review by rstijerina — APPROVED** [review]
Tested, works great! Just left a few comments with some suggested formatting.

**Pull request merged**